File: backend/src/main/java/com/interviewer/model/ProctorEvent.java
package com.interviewer.model;

import org.springframework.data.annotation.Id;
import org.springframework.data.mongodb.core.mapping.Document;
import java.time.LocalDateTime;

@Document(collection = "proctor_events")
public class ProctorEvent {

    @Id
    private String id;
    private String sessionId;
    private String eventType;
    private String severity;
    private String description;
    private LocalDateTime timestamp = LocalDateTime.now();

    public ProctorEvent() {}

    public String getId() { return id; }
    public void setId(String id) { this.id = id; }
    public String getSessionId() { return sessionId; }
    public void setSessionId(String sessionId) { this.sessionId = sessionId; }
    public String getEventType() { return eventType; }
    public void setEventType(String eventType) { this.eventType = eventType; }
    public String getSeverity() { return severity; }
    public void setSeverity(String severity) { this.severity = severity; }
    public String getDescription() { return description; }
    public void setDescription(String description) { this.description = description; }
    public LocalDateTime getTimestamp() { return timestamp; }
    public void setTimestamp(LocalDateTime timestamp) { this.timestamp = timestamp; }
}

File: backend/src/models/ProctorEvent.js
/**
 * ProctorEvent Model — Mongoose schema for proctor_events collection
 * Mirrors the Java ProctorEvent entity exactly.
 */

const mongoose = require('mongoose');

const proctorEventSchema = new mongoose.Schema({
  sessionId: {
    type: String,
    required: true
  },
  eventType: {
    type: String,
    required: true
  },
  severity: {
    type: String,
    default: 'MEDIUM'
  },
  description: {
    type: String,
    default: ''
  },
  timestamp: {
    type: Date,
    default: Date.now
  }
}, {
  collection: 'proctor_events'
});

module.exports = mongoose.model('ProctorEvent', proctorEventSchema);
